drake/systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -83,7 +84,17 @@
 kinsol.v = v;
 
 if (options.use_mex && model.mex_model_ptr~=0 && isnumeric(q))
-  doKinematicsmex(model.mex_model_ptr,q,options.compute_gradients,v,options.compute_JdotV);
+  if isempty(options.kinematics_cache_ptr_to_use)
+    if options.compute_gradients
+      kinsol.mex_ptr = model.default_kinematics_cache_ptr_with_gradients;
```

Comment:
right, i think that we would need a cache (of size one) so that it returns the same kinsol ptr if doKinematics is called with exactly the same arguments.  We’ve used that as the contract everywhere — it’s ok to call doKinematics locally because it will return quickly after the first hit.

I’m ok accepting as is.

On Sep 4, 2015, at 3:14 PM, Twan Koolen <notifications@github.com> wrote:

> In drake/systems/plants/@RigidBodyManipulator/doKinematics.m:
> 
> > @@ -83,7 +84,17 @@
> >  kinsol.v = v;
> >  
> >  if (options.use_mex && model.mex_model_ptr~=0 && isnumeric(q))
> > -  doKinematicsmex(model.mex_model_ptr,q,options.compute_gradients,v,options.compute_JdotV);
> > +  if isempty(options.kinematics_cache_ptr_to_use)
> > +    if options.compute_gradients
> > +      kinsol.mex_ptr = model.default_kinematics_cache_ptr_with_gradients;
> Fixing things on the Matlab side is not the main objective of this pull request. In fact, I actively tried to avoid making major changes in Matlab, I just want to keep it working as is. Requiring that the caller passes in an existing one means I have to change all usages of doKinematics in Matlab and find a place to store a kinematics cache to reuse. Creating a new DrakeMexPointer to KinematicsCache makes everything much slower.
> 
> —
> Reply to this email directly or view it on GitHub.
>